src/Hl7.Fhir.Patch/Internal/ElementNodeAdapter.cs

```diff
@@ -0,0 +1,174 @@
+﻿/* 
+ * Copyright (c) 2020, Firely (info@fire.ly) and contributors
+ * See the file CONTRIBUTORS for details.
+ * 
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://github.com/FirelyTeam/fhir-net-api/blob/master/LICENSE
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Specification;
+
+namespace Hl7.Fhir.Patch.Internal
+{
+    /// <summary>
+    /// This API supports infrastructure and is not intended to be used
+    /// directly from your code. This API may change or be removed in future releases.
+    /// </summary>
+    internal class ElementNodeAdapter : IAdapter
+    {
+        public virtual bool TryAdd(
+            object target,
+            string name,
+            IStructureDefinitionSummaryProvider contractResolver,
```

Comment:
Maybe provide the `IStructureDefinitionSummaryProvider` just once in the constructor?